km_api/account/serializers.py

```diff
@@ -197,9 +197,11 @@ def validate(self, data):
                 associated with the confirmation.
         """
         confirmation = models.EmailConfirmation.objects.get(key=data['key'])
-        user = authenticate(
-            email=confirmation.email.email,
-            password=data['password'])
+
+        email = confirmation.email.email
+        user = models.User.objects.get(email=email)
+
+        assert user.check_password(data['password']), user is None
 
         if not user:
```

Comment:
This line should be where the password check is done. We know `user` is not `None` because we pulled it from the email instance, so this check is currently not useful.

It can be fixed by replacing this line with:

```python
if not user.check_password(data['password']):
```